benches/engines/mod.rs

```diff
@@ -0,0 +1,232 @@
+                config.value_length,
+            )
+        }
+        EngineType::RocksDB => {
+            let engine = TestEngineBuilder::new().build().unwrap();
+            engine_snapshot_bench(
+                bencher,
+                &engine,
+                config.engine_keys_count,
+                config.value_length,
+            )
+        }
+    }
+}
+
+#[derive(Debug)]
+struct GetConfig {
+    engine_type: EngineType,
+    iterations: usize,
+    value_length: usize,
+    engine_keys_count: usize,
+}
+
+fn bench_engine_get(bencher: &mut Bencher, config: &GetConfig) {
+    match config.engine_type {
+        EngineType::BTreeEngine => {
+            let engine = BTreeEngine::default();
+            engine_get_bench(
+                bencher,
+                &engine,
+                config.iterations,
+                config.engine_keys_count,
+                config.value_length,
+            )
+        }
+        EngineType::RocksDB => {
+            let engine = TestEngineBuilder::new().build().unwrap();
+            engine_get_bench(
+                bencher,
+                &engine,
+                config.iterations,
+                config.engine_keys_count,
+                config.value_length,
+            )
+        }
+    }
+}
+
+fn bench_engines(c: &mut Criterion) {
+    let engine_types = vec![EngineType::BTreeEngine, EngineType::RocksDB];
+    let value_lengths = vec![128, 1024];
+    let engine_keys_counts = vec![0, 1000, 10_000];
+    let engine_put_keys_counts = vec![0, 1000];
+
+    let mut get_configs = vec![];
+    let mut put_configs = vec![];
+    let mut snapshot_configs = vec![];
+
+    for engine_type in engine_types {
+        for &value_length in &value_lengths {
+            for &engine_keys_count in &engine_keys_counts {
+                get_configs.push(GetConfig {
+                    engine_type: engine_type.clone(),
+                    iterations: DEFAULT_ITERATIONS,
+                    value_length,
+                    engine_keys_count,
+                });
+                snapshot_configs.push(SnapshotConfig {
+                    engine_type: engine_type.clone(),
+                    value_length,
+                    engine_keys_count,
+                });
+            }
+
+            for &engine_put_keys_count in &engine_put_keys_counts {
+                put_configs.push(PutConfig {
+                    engine_type: engine_type.clone(),
+                    put_count: engine_put_keys_count,
+                    value_length,
+                });
+            }
+        }
+    }
+
+    c.bench_function_over_inputs("bench_engine_get", bench_engine_get, get_configs);
+    c.bench_function_over_inputs("bench_engine_put", bench_engine_put, put_configs);
+    c.bench_function_over_inputs(
+        "bench_engine_snapshot",
+        bench_engine_snapshot,
+        snapshot_configs,
+    );
+}
+
+fn main() {
+    let mut criterion = Criterion::default().sample_size(10);
```

Comment:
Engine get, put and snapshot should be very low-cost operations, so maybe it is not necessary to set `sample_size(10)` (if total running time is acceptable)?

Further, if `sample_size(10)` will be removed, then the total `fn main()` can be replaced by a more standard and recommended style: using `criterion_group` and `criterion_main`.

Reply:
`engine_put()` will put  1000 keys into engine per iteration, it cost too much time by default sample size, and I remain it here at last.